### PR TITLE
Move provider class manager interfaces to a public package

### PR DESCRIPTION
**Pull request by JAORMX**
# Summary

These are to be implemented by providers. Thus, they make no sense to be
internal interfaces.

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [ ] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [x] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [ ] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

**Comment by jhrozek** [comment]
the unit test failures look related to the PR

**Comment by JAORMX** [comment]
@jhrozek they are, I gotta look into them.

File: internal/controlplane/handlers_oauth.go [hunk 1]
```diff
@@ -46,9 +46,9 @@ import (
 	"github.com/mindersec/minder/internal/providers"
 	"github.com/mindersec/minder/internal/providers/credentials"
 	"github.com/mindersec/minder/internal/providers/github/service"
-	"github.com/mindersec/minder/internal/providers/manager"
 	"github.com/mindersec/minder/internal/util"
 	pb "github.com/mindersec/minder/pkg/api/protobuf/go/minder/v1"
+	mgrif "github.com/mindersec/minder/pkg/providers/v1/manager"
```
Comment (evankanderson) [inline internal/controlplane/handlers_oauth.go:51]: Why not keep the name `manager`?

File: internal/controlplane/server.go [hunk 2]
```diff
@@ -77,6 +77,7 @@ import (
 	"github.com/mindersec/minder/internal/ruletypes"
 	"github.com/mindersec/minder/internal/util"
 	pb "github.com/mindersec/minder/pkg/api/protobuf/go/minder/v1"
+	mgrif "github.com/mindersec/minder/pkg/providers/v1/manager"
```
Comment (evankanderson) [inline internal/controlplane/server.go:80]: Ditto -- `mgrif` seems less clear than `manager`.

File: internal/providers/github/manager/manager.go [hunk 3]
```diff
@@ -40,9 +40,9 @@ import (
 	"github.com/mindersec/minder/internal/providers/github/clients"
 	"github.com/mindersec/minder/internal/providers/github/properties"
 	"github.com/mindersec/minder/internal/providers/github/service"
-	m "github.com/mindersec/minder/internal/providers/manager"
 	"github.com/mindersec/minder/internal/providers/ratecache"
 	v1 "github.com/mindersec/minder/pkg/providers/v1"
+	mgrif "github.com/mindersec/minder/pkg/providers/v1/manager"
```
Comment (evankanderson) [inline internal/providers/github/manager/manager.go:45]: (And here -- we don't have a `manager` import already)

**Pull request closed without merging**